lib/ansible/module_utils/kubevirt.py

```diff
@@ -215,16 +221,23 @@ def _define_cloud_init(self, cloud_init_nocloud, template_spec):
                 'disk': {'bus': 'virtio'},
             })
 
-    def _define_interfaces(self, interfaces, template_spec):
+    def _define_interfaces(self, interfaces, template_spec, defaults):
         """
         Takes interfaces parameter of Ansible and create kubevirt API interfaces
         and networks strucutre out from it.
         """
+        if not interfaces and defaults and 'interfaces' in defaults:
+            interfaces = copy.deepcopy(defaults['interfaces'])
+            for d in interfaces:
+                d['network'] = defaults['networks'][0]
+
         if interfaces:
             # Extract interfaces k8s specification from interfaces list passed to Ansible:
             spec_interfaces = []
             for i in interfaces:
-                spec_interfaces.append(dict((k, v) for k, v in i.items() if k != 'network'))
+                spec_interfaces.append(
+                    dict(self.merge_dicts(dict((k, v) for k, v in i.items() if k != 'network'), defaults['interfaces']))
```

Comment:
Ah sorry.... you don't need to call ```dict()``` here anymore.  ```merge_dicts()``` is now returning a copy of the input dictionaries so you don't need to copy the copy here.  This can simply be:
``` python
spec_interfaces.append(
                    self.merge_dicts(dict((k, v) for k, v in i.items() if k != 'network'), defaults['interfaces'])
                    )
```

And the same for the other calls to ```merge_dicts()```

Reply:
Thanks, fixed.